reflect/src/main/java/dagger/reflect/UnlinkedJustInTimeBinding.java

```diff
@@ -75,7 +75,8 @@ private static boolean hasParameterizedTypeVariable(Type parameterType) {
   private TypeUtil.ParameterizedTypeImpl findKeyForParameterizedType(
       ParameterizedType parameterType) {
     Type[] matchingTypes = matchingParameterizedType(parameterType.getActualTypeArguments());
-    return new TypeUtil.ParameterizedTypeImpl(null, parameterType.getRawType(), matchingTypes);
+    return new TypeUtil.ParameterizedTypeImpl(
+        parameterType.getOwnerType(), parameterType.getRawType(), matchingTypes);
```

Comment:
@JakeWharton i wasn't passing in the owner type here which was causing the exception. it is now resolved.